packages/astro/src/core/build/static-build.ts

```diff
@@ -125,6 +148,22 @@ async function ssrBuild(opts: StaticBuildOptions, internals: BuildInternals, inp
 					assetFileNames: 'assets/[name].[hash][extname]',
 					...viteConfig.build?.rollupOptions?.output,
 					entryFileNames: opts.buildConfig.serverEntry,
+					manualChunks: opts.settings.config.experimental.prerender ? (id, api) => {
```

Comment:
I think this should be done in a plugin rather than in this file, to improve readability. Like in a new prerender plugin.

Reply:
Also I think another plugin is setting manualChunks, is this overriding that? I think in that plugin we try and check to see if manualChunks already exists and if so it call it first.

Reply:
Good call, didn't even consider that. Will update.

Reply:
Refactored into `vite-plugin-prerender` in [ecc44ab](https://github.com/withastro/astro/pull/5297/commits/ecc44ab042d17b640dda3f3e600d46f2812e203d)